cmd/ako-main/main.go

```diff
@@ -376,9 +378,17 @@ func istioWatcherEvents(watcher *fsnotify.Watcher, kc *kubernetes.Clientset, ist
 					watcher.Add(lib.IstioCertOutputPath)
 					utils.AviLog.Infof("added path to %s to istio watcher", lib.IstioCertOutputPath)
 					initIstioSecrets(kc, istioUpdateCh)
-
 				} else {
-					lib.CreateIstioSecretFromCert(event.Name, kc)
+					if strings.HasSuffix(event.Name, "cert-chain.pem") ||
+						strings.HasSuffix(event.Name, "key.pem") ||
+						strings.HasSuffix(event.Name, "root-cert.pem") {
+						lib.CreateIstioSecretFromCert(event.Name, kc)
+						if !lib.IsChanClosed(*istioUpdateCh) {
+							if lib.GetIstioCertSet().Len() == 3 {
```

Comment:
are we accessing the global variable from two different go routines? that may cause an issue.